test/Microsoft.Azure.WebJobs.Host.EndToEndTests/SingletonEndToEndTests.cs

```diff
@@ -652,6 +656,9 @@ private IHost CreateTestJobHost<TProg>(int hostId, Action<IHostBuilder> extraCon
                         o.LockAcquisitionTimeout = TimeSpan.FromSeconds(10);
                         o.LockAcquisitionPollingInterval = TimeSpan.FromMilliseconds(500);
                     });
+
+                    services.AddSingleton<IDistributedLockManager, BlobLeaseDistributedLockManager>();
+                    services.AddSingleton<IAzureStorageProvider, TestAzureStorageProvider>();
```

Comment:
As we discussed, this shouldn't be needed - AddAzureStorageCoreServices should add everything required. These E2E tests are verifying the end user experience and users shouldn't have to implement IDistributedLockManager/IAzureStorageProvider